micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java

```diff
@@ -203,11 +203,11 @@ private boolean isConcurrentPhase(String cause) {
     }
 
     private boolean isOldGenPool(String name) {
-        return name.endsWith("Old Gen") || name.endsWith("Tenured Gen");
+        return name.endsWith("Old Gen") || name.endsWith("Tenured Gen") || name.equals("tenured-SOA");
```

Comment:
If I am reading the referenced documentation right, "tenured-LOA" should also be considered. As objects are either moved from "nursery" to SOA or LOA if the exceed a certain size. Please correct me if I am wrong.

> Within the tenure area, new objects are allocated into the small object area (SOA), which is illustrated in the earlier diagram (see 3.). A large object area (LOA) is set aside for objects greater than 64 KB that cannot be allocated into the SOA to minimize fragmentation.

This would mean though, that we just can't keep the 'oldGenPoolName' and use it, but rather query more then pool name from the 'GCInfo.getMemoryUsage*Gc()' maps on every garbage collection notification. (Again, under the assumption the both, SOA and LOA make up the metric for "promoted" bytes.)

Reply:
Yes you are right of course. I just read the LOA area since I did not like to rewrite the whole class. LOA is more important, actually in my applications (average spring boot stuff) SOA is 0 bytes...
To be 100% correct we would have to consider multiple old gen pools.

I'll see if I can implement this without changing to much.